Sources/Sentry/SentryThreadHandle.cpp

```diff
@@ -128,21 +128,50 @@ namespace profiling {
     ThreadHandle::dispatchQueueAddress() const noexcept
     {
         if (handle_ == THREAD_NULL) {
-            return {};
+            return 0;
         }
+
         integer_t infoBuffer[THREAD_IDENTIFIER_INFO_COUNT] = { 0 };
         thread_info_t info = infoBuffer;
         mach_msg_type_number_t count = THREAD_IDENTIFIER_INFO_COUNT;
+        if (thread_info(handle_, THREAD_IDENTIFIER_INFO, info, &count) != KERN_SUCCESS) {
+            return 0;
+        }
+        if (info == nullptr) {
```

Comment:
`info` isn't a pointer, it's a stack allocated struct, what does the nullptr check do?

Reply:
`info`'s type is `thread_info_t` which is typedef'd to `integer *`, so i thought maybe...

Reply:
Ah, I see. Still quite unlikely if `thread_info` doesn't return a non-success code though.

Reply:
Technically yes, it's a pointer, but on line 135 set `info` to point to memory allocated on the stack. The only way this could be `nullptr` is if `info` was passed by reference to `thread_info` and `thread_info` set it to `nullptr`, but it is not passed by ref in this case.